py3status/modules/i3pystatus.py

```diff
@@ -0,0 +1,328 @@
+
+        # backends
+        backend_type = None
+        if 'backend' in settings:
+            backend_type = 'backend'
+        elif 'backends' in settings:
+            backend_type = 'backends'
+
+        if backend_type:
+            backends = settings[backend_type]
+            backends_initiated = []
+            for key, value in backends.items():
+                mod_info = key.split('.')
+                backend_module = import_module(get_backends().get(mod_info[0]))
+                try:
+                    backend = getattr(backend_module, mod_info[1])(**value)
+                except i3pystatus.core.exceptions.ConfigMissingError as e:
+                    msg = e.message
+                    msg = (
+                        'i3pystatus module `{}` backend `{}`'
+                        'missing configuration options {}'
+                    ).format(
+                        self.module, key, msg[msg.index('{'):]
+                    )
+                    self.py3.notify_user(msg)
+                    raise Exception('Missing configuration options')
+                backends_initiated.append(backend)
+            if backend_type == 'backend':
+                settings['backend'] = backends_initiated[0]
+            else:
+                settings['backends'] = backends_initiated
+
+        # i3pystatus.Status can only exist once
+        # so create it and share
+        global status
+
+        if status is None:
+            status = i3pystatus.Status()
+
+        # create the module and register it
+        finder = status.modules.finder
+        try:
+            module = finder.instanciate_class_from_module(self.module, **settings)
+        except i3pystatus.core.exceptions.ConfigMissingError as e:
+            msg = e.message
+            msg = 'i3pystatus module `{}` missing configuration options {}'.format(
+                self.module, msg[msg.index('{'):]
+            )
+            self.py3.notify_user(msg)
+            raise Exception('Missing configuration options')
+        status.register(module)
+        self.module = module
+
+        self.is_interval_module = isinstance(module, i3pystatus.IntervalModule)
+        # modules update their output independently so we need to periodically
+        # check if it has been updated.  For modules with long intervals it is
+        # important to do this output check much more regularly thank the
+        # interval.
+        self._cache_timeout = min(
+            getattr(module, 'interval', MIN_CHECK_INTERVAL), MIN_CHECK_INTERVAL
+        )
+
+        # get callbacks available, useful for deciding if double clicks exist
+        callbacks = []
+        for i in range(len(CLICK_EVENTS)):
+            click = getattr(module, CLICK_EVENTS[i], None)
+            dbclick = getattr(module, DBL_CLICK_EVENTS[i], None)
+            callbacks.append((click, dbclick))
+
+        self._click_timer = ClickTimer(self, callbacks)
+        self._due_time = 0
```

Comment:
(Q) Purpose for this config?

Reply:
I think this is redundant I need to do more testing of this module